packages/code-du-travail-ui/src/Tabs/index.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import { Tab, Tabs as RootTabs, TabList, TabPanel } from "react-tabs";
+import { box, colors, fonts, spacing } from "../theme";
+
+class Tabs extends React.PureComponent {
+  render() {
+    const { data, defaultIndex, onSelect, selectedIndex } = this.props;
+    const tabs = data.map(item => <Tab key={item.key}>{item.tab}</Tab>);
+    const tabContents = data.map(item => (
+      <TabPanel key={item.key}>{item.panel}</TabPanel>
+    ));
+
+    const props = {
+      onSelect,
+      // conditional props
+      ...(typeof selectedIndex === "number" && { selectedIndex }),
+      ...(isNaN(selectedIndex) && { defaultIndex })
+    };
+
+    return (
+      <StyledTabs {...props}>
+        <TabList>{tabs}</TabList>
+        {tabContents}
+      </StyledTabs>
+    );
+  }
+}
+
+Tabs.propTypes = {
+  withHistory: PropTypes.bool,
+  onSelect: PropTypes.func,
+  selectedIndex: PropTypes.number,
+  defaultIndex: PropTypes.number,
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      tab: PropTypes.node.isRequired,
+      panel: PropTypes.node.isRequired
+    })
+  ).isRequired
+};
+
+Tabs.defaultProps = {
+  withHistory: false,
```

Comment:
this is interesting :) does it works ? why default it to false ?

Reply:
Oh I forgot to remove that one ^^ At first I thought I could handle the push state at the component level but in the end I think it is better to let the parent component handle it if it wishes so.